sdk/identity/azure_identity/src/credentials/chained_credentials.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use crate::credentials::cache::TokenCache;
+use azure_core::{
+    credentials::{AccessToken, TokenCredential},
+    error::{Error, ErrorKind},
+};
+use std::sync::Arc;
+
+/// Provides a mechanism of selectively adding credentials used for a `ChainedTokenCredential` instance
+#[derive(Default)]
+pub struct ChainedTokenCredentialBuilder {
```

Comment:
We don't use builders anymore. All options should be plain old objects per [guidelines](https://azure.github.io/azure-sdk/rust_introduction.html#rust-client-configuration).